backend/corpora/common/entities/dataset.py

```diff
@@ -145,31 +150,43 @@ def dataset_and_asset_deletion(self):
         for artifact in self.artifacts:
             asset = DatasetAsset.get(self.session, artifact.id)
             asset.delete_from_s3()
+        self.delete_deployment_directories()
         self.tombstone_dataset_and_delete_child_objects()
 
+    def delete_deployment_directories(self):
+        for deployment_directory in self.deployment_directories:
+            object_names = self.get_cxg_bucket_path(deployment_directory)
+            logger.info(f"Deleting all files in bucket {cxg_bucket.name} under {object_names}.")
+            cxg_bucket.objects.filter(Prefix=object_names).delete()
+
+    @staticmethod
+    def get_cxg_bucket_path(deployment_directory: str) -> str:
+        """Parses the S3 cellxgene bucket object prefix for all resources related to this dataset from the
+        deployment directory URL"""
+        object_name = urlparse(deployment_directory.url).path.split("/", 2)[2]
+        base, _ = os.path.splitext(object_name)
+        return base
+
     @staticmethod
-    def new_processing_status():
+    def new_processing_status() -> dict:
         return {
             "upload_status": UploadStatus.WAITING,
             "upload_progress": 0,
             "processing_status": ProcessingStatus.PENDING,
         }
 
-    def copy_csv_to_s3(self, csv_file):
-        s3 = boto3.resource("s3", endpoint_url=os.getenv("BOTO_ENDPOINT_URL"))
-        cellxgene_bucket = os.getenv("CELLXGENE_BUCKET", f"hosted-cellxgene-{os.environ['DEPLOYMENT_STAGE']}")
-        s3.meta.client.upload_file(csv_file, cellxgene_bucket, csv_file)
+    def copy_csv_to_s3(self, csv_file: str) -> str:
+        s3_file = f"{self.get_cxg_bucket_path(self.deployment_directories[0])}/genesets.csv"
+        cxg_bucket.upload_file(csv_file, s3_file)
+        return s3_file
 
-    def generate_tidy_csv_for_all_linked_genesets(self, csv_file=None):
-        if csv_file is None:
-            base, ext = os.path.splitext(self.deployment_directories[0].url)
-            if ext is not None:  # strip extension, if any
-                csv_file = f"{base}-genesets.csv"
+    def generate_tidy_csv_for_all_linked_genesets(self, csv_file_path: str) -> str:
+        csv_file = os.path.join(csv_file_path, "geneset.csv")
```

Comment:
hardcoding `geneset.csv` name, and combining it with the file path passed in as the `csv_file_path` parameter.